smack-extensions/src/test/java/org/jivesoftware/smackx/pubsub/ConfigureFormTest.java

```diff
@@ -95,12 +96,14 @@ public void getConfigFormWithTimeout() throws XMPPException, SmackException, Int
         DiscoverInfo discoverInfo = info.build();
         con.addIQReply(discoverInfo);
 
-        Node node = mgr.getNode("princely_musings");
+        assertThrows(Exception.class, () -> {
```

Comment:
Should be `SmackException` that is expected to be thrown here. And that by `getNodeConfiguration()` I'd assume, hence the lambda should be smaller.

Reply:
Thanks for pointing that out. However, `getNode("pricely_musings");` has the potential to throw a `NotAPubSubNodeException` and hence I am keeping it inside the lambda. If this is not the case, please let me know and I'll make changes accordingly.

Reply:
Testing for expected exceptions is not about which exceptions could potentially be thrown, but which are actually thrown. So you test that, what is expected, actually happens, everything unforeseen happening is a violation of the test spec and should make the test fail.

Reply:
Got it !

Reply:
@Flowdalic 
Here, I think in master, the expected `SmackException` is raised by `mgr.getNode("princely_musings");` and not by `node.getNodeConfiguration();`.
This leaves the following code snippet unreachable and smoothly pass the test.
    `SmackConfiguration.setDefaultReplyTimeout(100);
     con.setTimeout();
     node.getNodeConfiguration();`.

Reply:
A `TODO` has been added.